controlplane/kubeadm/controllers/remediation.go

```diff
@@ -37,6 +37,29 @@ import (
 func (r *KubeadmControlPlaneReconciler) reconcileUnhealthyMachines(ctx context.Context, controlPlane *internal.ControlPlane) (ret ctrl.Result, retErr error) {
 	log := ctrl.LoggerFrom(ctx)
 
+	// Cleanup pending remediation actions not completed for any reasons (e.g. number of current replicas is less or equal to 1)
+	// if the underlying machine is now back to healthy / not deleting.
+	healthyMachines := controlPlane.HealthyMachines()
+	for _, m := range healthyMachines {
+		if conditions.IsTrue(m, clusterv1.MachineHealthCheckSuccededCondition) && conditions.IsFalse(m, clusterv1.MachineOwnerRemediatedCondition) && m.DeletionTimestamp.IsZero() {
+			patchHelper, err := patch.NewHelper(m, r.Client)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+
+			conditions.Delete(m, clusterv1.MachineOwnerRemediatedCondition)
+
+			if err := patchHelper.Patch(ctx, m, patch.WithOwnedConditions{Conditions: []clusterv1.ConditionType{
+				clusterv1.MachineOwnerRemediatedCondition,
+			}}); err != nil {
+				log.Error(err, "Failed to patch control plane Machine", "machine", m.Name)
+				if retErr == nil {
+					retErr = errors.Wrapf(err, "failed to patch control plane Machine %s", m.Name)
+				}
```

Comment:
Do we need to log twice?

Reply:
Same question for l.96

Reply:
Fixed,
L96 is a defer statement, so it is different from this loop. this could be fixed in a separated PR